[기능추가] 리액트쿼리로 만든 커스텀 훅들을 추가했습니다

porapo99 opened the pull request:
[코드 기능]

test-page에 각 분류별로 테스트훅을 테스트할수있는 페이지가 있습니다.

hooks 폴더에는 스웨거의 분류대로 각종 데이터 패칭에 관한 코드들이 있습니다.

api/proxy 파일에는 현재 외부의 url에 직접적으로 데이터통신을 하면 위험하다는 cors 오류를 해결하기위한 데이터통신을 next.js의 api기능을이용해 해결하였으며 자세하게 활용하는법은 잘 모르겠으니. 멘토님께 질문할 예정입니다.

util 함수로 쿠키를 저장하고 쿠키를 가져오는 함수를 만들어서 axios.instance 에 그걸 활용하게 코드를 만들었습니다.
( 쿠키를 헤더에 보내지않으면 아무런 데이터 패칭도 작동하지않습니다. )

[테스트 결과]
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/24d0e98c-d8ba-4ad7-8bef-5346257ad55d)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/524f83c1-7601-403e-998c-43dee67058dd)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/b9476e5d-ef89-42f6-b0e0-831cc21e1e95)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/1ed15f44-0ba1-42c5-99f4-3d3c2712e1d2)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/403355f5-a098-43e4-b952-0efe7d15f96e)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/b194c224-f7a9-44e1-b8bf-1a69bc108510)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/8ff3f391-a6a0-4896-8cfb-02904d97c8c7)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/8644944c-b5be-4928-8e66-5b53df1285e7)
![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/fe271a19-6f73-4fbb-94b0-1b924cdf383d)

porapo99 committed:
feat  : instance 추가

porapo99 committed:
fix : axios instance를 수정하였습니다.

porapo99 committed:
feat : errorHandler

porapo99 committed:
Merge branch 'feature/datafetch' of https://github.com/codeit-part4-project/Mogazoa into feature/datafetch

foxholic9 committed:
Feat: fetch 응답 타입 작성

현재 단순히 Swagger에서 복붙해서 작성해서 코드가 긴 편인데, 필요하면 공통부분 extends 문법으로 수정하겠습니다.

전 프로젝트에서 이미지 url 타입은 string으로 사용한게 편했던거같아서 string 사용했습니다.

순서는 Swagger문서 순이고, 여러 api에서 동일한 응답이 올 경우 주석 추가했습니다.

foxholic9 committed:
Feat: fetch 요청 바디 타입 작성

Date 타입을 string으로 수정,
작명을 response와 request로 일관화.
주석 추가
swagger내에서 테스트 해본 상태입니다.

porapo99 committed:
feat : use쿼리를 활용한 커스텀훅을 만들고있습니다.

현재 오류가있는 상태입니다.

porapo99 committed:
fix : 오류수정

@tenstack을 추가하지않아 리액트쿼리가 작동하지않던 오류를 추가하며 common.ts에 파일들을 수정하였습니다.

porapo99 committed:
feat : 모든 커스텀 훅 기능 작성

porapo99 committed:
test : fetch 테스트용 브랜치 커밋입니다.

porapo99 committed:
feat : test용 페이지 추가 및 코드의 안쓰던 쓸때없는 부분 삭제

porapo99 committed:
fix : 간단한 타입 버그를 수정했습니다

porapo99 committed:
feat : 회원가입도 쿠키를 추가하는 코드를 추가했습니다

porapo99 committed:
fix : env 사용하게 픽스했습니다.

porapo99 committed:
fix : 테스트용으로 쓰던 console.log 삭제

porapo99 commented:
TODO : 프록시 서버 쓰면서 뭐 때문인지 모를 TeamId가 proxy 라는 단어로 수정된점 고쳐야됨.
연달아서 작업하게 되면서 코드가 더럽고 길어짐 리팩토링이 필요할 확률 매우높음.

porapo99 commented on app/test-page/auth-test/page.tsx:
```diff
@@ -0,0 +1,166 @@
+        setCookie('accessToken', data.accessToken, 7);
+      }
+    },
+  });
```
커스텀훅에 option을 사용한 모습

porapo99 committed:
fix : 테스트용 폼에 경고 수정

porapo99 committed:
cleanup : 주석 삭제 , 사용하지않는 타입들 정리.

porapo99 committed:
Merge branch 'feature/fetch' of https://github.com/codeit-part4-project/Mogazoa into feature/fetch

porapo99 committed:
fix : teamId가 proxy로 나오던 문제 해결

추가로 불필요한 타입들 조금 손봤습니다.

porapo99 committed:
fix : 안쓰는 타입 삭제

choinashil commented on app/api/proxy/[...path]/route.ts:
```diff
@@ -0,0 +1,85 @@
+
+      try {
+        const jsonBody = JSON.parse(body);
+        if (jsonBody.teamId) {
+          jsonBody.teamId = '5-6';
```
body 에 teamId 가 필요한가요? api 문서 봤을 때는 보통 path variable 로 전달하는 것 같아서요~

choinashil commented on app/api/proxy/[...path]/route.ts:
```diff
@@ -0,0 +1,85 @@
+
+  const headers = new Headers(request.headers);
+
+  headers.delete('Origin');
```
Origin 삭제 이유는 무엇 때문일까요?

choinashil commented on hooks/auth.ts:
```diff
@@ -0,0 +1,48 @@
+  const method = 'post';
+  const url = '/auth/signUp';
+  const mutation = useDataMutation<SignUpRequest, AuthResponse>(url, method, options);
+  return (data: SignUpRequest) => mutation.mutate(data);
```
여기서 mutate 하는 함수를 리턴하면 mutate 시 isLoading 이나 error 등은 참고하기 까다로울 것 같네요
(isLoading, error 등은 mutation 에 들어있음)

choinashil commented on services/common.ts:
```diff
@@ -0,0 +1,78 @@
+    mutationFn: (data: TData | null) =>
+      fetchData<TData, TResponse>({ url, method, data, headers: options?.headers }),
+    onError: (error) => {
+      const errorMessage = handleErrorResponse(error);
+      alert(errorMessage);
```
에러 발생 시 공통으로 alert 되게끔 처리하셨는데 개별 상황마다 에러 처리를 다르게 해야 하는 경우 (alert 가 노출되지 않았으면 좋겠다거나) 도 고려해서 개선해보시면 좋을 것 같습니다!

choinashil commented on services/common.ts:
```diff
@@ -0,0 +1,78 @@
+  };
+
+  if (method !== 'get') {
+    if (headers?.['Content-Type'] === 'multipart/form-data') {
+      config.data = data;
+    } else {
+      config.data = data;
```
두 로직 간의 차이가 뭔지 모르겠네요!

choinashil commented on utils/axiosInstance.ts:
```diff
@@ -0,0 +1,31 @@
+instance.interceptors.response.use(
+  (response) => {
+    if (response.data && response.data.accessToken) {
+      setCookie('accessToken', response.data.accessToken, 7);
```
상수는 분리해서 관리하는 것을 추천드립니다!

porapo99 committed:
fix : 피드백 주신것들 수정했습니다.

porapo99 committed:
[기능수정] 데이터패칭 훅들 리팩토링 (#12)

이전과 뮤테이션 훅 사용할때 사용법이 조금 달라졋습니다.
방법은 크게 다르지않아서 test-page에 예시 남겨놨습니다.
사용법 전부 주석 달아두겠습니다.

테스트결과는 이전과 다르지않기에 생략합니다.

porapo99 committed:
[기능수정] : teamId가 proxy로 나오던 문제 해결 (#11)

teamid가 http://localhost:3000/api/proxy/auth/signUp 로 통신되었기떄문에 proxy로
나오던거같아서 아예 폴더이름을 5-6로 바꾼다음에 통신하였습니다. 그리고 원래 시도했던 방식이 안됐기에 시도했던 불필요한 코드들
삭제하였습니다

추가로 불필요한 타입들 조금 손봤습니다.

[테스트결과]


![image](https://github.com/codeit-part4-project/Mogazoa/assets/151392186/9660390c-bb99-4258-8cfe-164e914c0742)

porapo99 committed:
fix : 버젼이 잘못 받아진 것을 수정하였습니다.
